src/JuMP.jl

```diff
@@ -203,10 +203,26 @@ function Base.copy(source::Model)
     # Variables
     dest.numCols = source.numCols
     dest.colNames = source.colNames[:]
+    dest.colNamesIJulia = source.colNamesIJulia[:]
     dest.colLower = source.colLower[:]
     dest.colUpper = source.colUpper[:]
     dest.colCat = source.colCat[:]
 
+    # callbacks and hooks
+    dest.callbacks = copy(source.callbacks)
```

Comment:
Since callbacks are wrapped in types, not plain functions, I think this will result in a no method error, except if a user explicitly defines a copy method for their custom callback.